pkg/kubelet/prober/prober_manager.go

```diff
@@ -236,19 +236,33 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 			started = !exists
 		}
 		podStatus.ContainerStatuses[i].Started = &started
-
 		if started {
 			var ready bool
 			if c.State.Running == nil {
 				ready = false
-			} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok && result == results.Success {
-				ready = true
+			} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok {
+				if result == results.Success {
+					ready = true
+				} else {
+					ready = false
```

Comment:
Isn't it already initialized at `false`?

Reply:
You are right, this could be improved and is maybe leftover from development.